lua/cfc_prop_restoration/core/sv/sv_adinterface.lua

```diff
@@ -0,0 +1,146 @@
+    end
+
+    local sortingSystems = {}
+    local fullSystems = {}
+    local function buildSystems( input )
+        while next( input ) ~= nil do
+            for k, v in pairs( input ) do
+                for systemi, system in pairs( sortingSystems ) do
+                    for _, target in pairs( system ) do
+                        for x = 1, 4 do
+                            if v.Entity[x] then
+                                for y = 1, 4 do
+                                    if target.Entity[y] and v.Entity[x].Index == target.Entity[y].Index then
+                                        system[#system + 1] = v
+                                        if #system == 100 then
+                                            fullSystems[#fullSystems + 1] = system
+                                            table.remove( sortingSystems, systemi )
+                                        end
+                                        input[k] = nil
+                                        goto super_loopbreak
+                                    end
+                                end
+                            end
+                        end
+                    end
+                end
+            end
+
+            --Normally skipped by the goto unless no cluster is found. If so, make a new one.
+            local k = next( input )
+            sortingSystems[#sortingSystems + 1] = { input[k] }
+            input[k] = nil
+
+            ::super_loopbreak::
+        end
+    end
+
+    buildSystems( sorted )
+    buildSystems( nocollide )
+
+    local ret = {}
+    for _, system in pairs( fullSystems ) do
+        for _, v in pairs( system ) do
+            ret[#ret + 1] = v
+        end
+    end
+
+    for _, system in pairs( sortingSystems ) do
+        for _, v in pairs( system ) do
+            ret[#ret + 1] = v
+        end
+    end
+
+    for k, v in pairs( unsorted ) do
+        ret[#ret + 1] = v
+    end
+
+    if #fullSystems ~= 0 then
+        ply:ChatPrint( "DUPLICATOR: WARNING, Number of constraints exceeds 100: (" .. #ret .. "). Constraint sorting might not work as expected." )
+    end
+
+    return ret
+end
+
+local function copyPlayerProps( ply )
+    --select all owned props
+    local entities = {}
+    for _, ent in pairs( ents.GetAll() ) do
+        if playerCanDupeCPPI( ply, ent ) then
+            entities[ent:EntIndex()] = ent
+        end
+    end
+
+    local ent = entities[next( entities )]
+    local headEnt = {}
+    headEnt.Index = ent:EntIndex()
+    headEnt.Pos = ent:GetPos()
+
+    local entities, constraints = AdvDupe2.duplicator.AreaCopy( entities, headEnt.Pos, true )
+    
+    return { entities, constraints, headEnt }
+end
+
+local function pastePlayerProps( ply, data )
+    local entities, constraints, headEnt = unpack( data )
+
+    if ply.AdvDupe2.Pasting or ply.AdvDupe2.Downloading then
+        AdvDupe2.Notify( ply, "Advanced Duplicator 2 is busy.", NOTIFY_ERROR )
+        return false
+    end
```

Comment:
Do we need any additional error handling here? Maybe a `logger:error` or something so it sends a non-halting error?

Reply:
No that's nothing we really have to worry about.